hosters/hetzner-dedicated/hetzner-dedicated-wipe-and-install-nixos.sh

```diff
@@ -237,51 +336,76 @@ cat > /mnt/etc/nixos/configuration.nix <<EOF
   # We do not worry about plugging disks into the wrong machine because
   # we will never exchange disks between machines.
   environment.etc."mdadm.conf".text = ''
-    HOMEHOST hetzner
+    HOMEHOST $HOSTNAME
   '';
+
   # The RAIDs are assembled in stage1, so we need to make the config
   # available there.
   boot.initrd.mdadmConf = config.environment.etc."mdadm.conf".text;
 
   # Network (Hetzner uses static IP assignments, and we don't use DHCP here)
   networking.useDHCP = false;
-  networking.interfaces."$NIXOS_INTERFACE".ipv4.addresses = [
-    {
-      address = "$IP_V4";
-      prefixLength = 24;
-    }
-  ];
-  networking.interfaces."$NIXOS_INTERFACE".ipv6.addresses = [
-    {
-      address = "$IP_V6";
-      prefixLength = 64;
-    }
-  ];
-  networking.defaultGateway = "$DEFAULT_GATEWAY";
-  networking.defaultGateway6 = { address = "fe80::1"; interface = "$NIXOS_INTERFACE"; };
-  networking.nameservers = [ "8.8.8.8" ];
+
+  networking.interfaces."$NET_IFACE" = {
+    ipv4 = {
+      addresses = [{
+        # Server main IPv4 address
+        address = "$IP_V4";
+        prefixLength = 24;
+      }];
+
+      routes = [
+        # Default IPv4 gateway route
+        {
+          address = "0.0.0.0";
+          prefixLength = 0;
+          via = "$DEFAULT_GATEWAY";
+        }
+      ];
+    };
+
+    ipv6 = {
+      addresses = [{
+        address = "$IP_V6";
+        prefixLength = 64;
+      }];
+
+      # Default IPv6 route
+      routes = [{
+        address = "::";
+        prefixLength = 0;
+        via = "fe80::1";
+      }];
+    };
+  }
+
+  networking = {
+    nameservers = [ "8.8.8.8" "8.8.4.4" ];
+    hostName = "$HOSTNAME";
+    ${NET_SLAVE:-}
+  };
+
+  ${NET_SYSCTL:-}
 
   # Initial empty root password for easy login:
   users.users.root.initialHashedPassword = "";
   services.openssh.permitRootLogin = "prohibit-password";
+  services.openssh.enable = true;
 
   users.users.root.openssh.authorizedKeys.keys = [
-    "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAtwCIGPYJlD2eeUtxngmT+4yR7BMlK0F5kzj+84uHsxxsy+PXFrP/tScCpwmuoiEYNv/9WKnPJJfCA9XlIDr6cla1MLpaW6eg672TRYMmKzH6SLlkg+kyDmPxSIJw+KdKfnPYyva+Y/VocACYJo0voabUeLAVgtSKGz/AFzccjfOR0GmFO911zjAaR+jFb9M7t7dveNVKm9KbuBfu3giMgGg3/mKz1TKY8yk2ZOxpT5CllBb+B5BcEf+7IGNvNxr1Z0zz5cFXQ3LyBIZklnC/OaQCnD78BSiyPTkIXcmBFal2TaFwTDvki6PuCRpJy+dU1fDdgWLql97D0SVnjmmomw== nh2@deditus.de"
+    "$(echo "$SSH_PUBKEY" | xargs)"
```

Comment:
What is the reasoning behind `xargs` here? Just replace newlines with spaces and trim?

Reply:
Yep, exactly.